pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,376 @@
+                    enabled: this.kpatchService.enabled,
+                    justCurrent: current && !state.auto,
+                    applyCheckbox: current,
+                });
+            });
+        });
+    }
+
+    checkSetup() {
+        cockpit.file("/etc/dnf/plugins/kpatch.conf").read()
+                .then(data => {
+                    if (data) {
+                        const auto = /autoupdate\s*=\s*True/.test(data);
+                        this.setState((state, _) => {
+                            const current = state.enabled && (state.patchInstalled || state.patchUnavailable);
+                            return ({
+                                auto: !!auto,
+                                justCurrent: current && !auto,
+                                applyCheckbox: current,
+                            });
+                        });
+                    }
+                })
+                .catch(() => true); // Ignore errors, most likely just does not exist
+
+        cockpit.spawn(["uname", "-r"])
+                .then(data => {
+                    const fields = data.split(".");
+                    const sanitized_kernel_ver = fields.slice(0, fields.length - 2)
+                            .join("_");
+                    const patch_name = "kpatch-patch-" + sanitized_kernel_ver;
+                    check_missing_packages([patch_name])
+                            .then(d => {
+                                this.setState((state, _) => {
+                                    const installed = (d.unavailable_names || []).length === 0 && (d.missing_names || []).length === 0;
+                                    const unavailable = (d.unavailable_names || []).length > 0;
+                                    const current = state.enabled && (installed || unavailable);
+                                    return ({
+                                        patchName: patch_name,
+                                        patchInstalled: installed,
+                                        patchUnavailable: unavailable,
+                                        justCurrent: current && !state.auto,
+                                        applyCheckbox: current,
+                                        kernelName: data,
+                                    });
+                                });
+                            })
+                            .catch(() => true);
+                })
+                .catch(() => true);
+    }
+
+    handleInstall() {
+        this.setState({ updating: true });
+        install_dialog(this.state.missing)
+                .then(() => {
+                    this.setState({ missing: [], updating: false });
+                })
+                .catch(() => this.setState({ updating: false }));
+    }
+
+    onClose() {
+        this.setState((state, _) => {
+            const current = state.enabled && (state.patchInstalled || state.patchUnavailable);
```

Comment:
I've seen this "enabled && (installed || unavailable)" pattern at least four times now. Little helper property?